### PR TITLE
Update web-nfc nfc_push.https.html

### PR DESCRIPTION
**Pull request by xiuqijix**
part of #14389

@Honry, PTAL.

### CODE REVIEW

File: web-nfc/NFCWriter_push.https.html [hunk 1]
```diff
@@ -75,81 +75,98 @@
...
-  let promises = [];
+  const writer = new NFCWriter();
+  const promises = [];
   invalid_syntax_messages.forEach(message => {
     promises.push(
-      promise_rejects(t, 'SyntaxError', navigator.nfc.push(message)));
+      promise_rejects(t, 'SyntaxError', writer.push(message)));
   });
   return Promise.all(promises);
 }, "'Test that promise is rejected with SyntaxError if NDEFMessage contains invalid records.");
```
Comment (Honry) [inline web-nfc/NFCWriter_push.https.html]: Nit: s/NDEFMessage/NDEFMessageSource

File: web-nfc/NFCWriter_push.https.html [hunk 2]
```diff
@@ -75,81 +75,98 @@
...
 
 promise_test(t => {
-  return promise_rejects(t, new TypeError(), navigator.nfc.push(test_text_data, { timeout: "invalid"}));
-}, "nfc.push should fail with TypeError when invalid timeout is provided.");
+  const writer = new NFCWriter();
+  const controller = new AbortController();
+  const p = writer.push(test_text_data, { signal: controller.signal })
```
Comment (Honry) [inline web-nfc/NFCWriter_push.https.html]: Nit: add ";" at the end of the line.

File: web-nfc/NFCWriter_push.https.html [hunk 3]
```diff
@@ -75,81 +75,98 @@
...
 promise_test(t => {
-  let promise = navigator.nfc.push(test_text_data, { timeout: 100 });
-  navigator.nfc.cancelPush();
-  return promise_rejects(t, 'AbortError', promise);
-}, "nfc.cancelPush should reject pending promise with AbortError.");
+  const writer = new NFCWriter();
+  return promise_rejects(t, 'TimeoutError', writer.push(test_text_data, { timeout: 1 }));
+}, "NFCWriter.push should fail with TimeoutError when timer expires.");
 
 promise_test(t => {
```
Comment (Honry) [inline web-nfc/NFCWriter_push.https.html]: Nit: Could you move this test behind the other `Signal` test above? Just to make tests ordered. :)

**Review by Honry — APPROVED** [review]
Thank you @kaixinjxq, looks pretty good, only some nits.

**Comment by kaixinjxq** [comment]
@Honry, many thanks for your review! I just submit a new commit to address your comments, PTAL.

**Pull request merged**